policy/modules/system/init.if

```diff
@@ -3552,6 +3552,43 @@ interface(`init_manage_all_unit_files',`
 	manage_lnk_files_pattern($1, systemdunit, systemdunit)
 ')
 
+########################################
+## <summary>
+##	Create a specific unit filetype in unit directory
+##	The desired use case here is for runtime unit files
+##	in /run/systemd/system but understanding this will
+##	also allow creation in /etc/systemd/system and others
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <param name="private type">
+##	<summary>
+##	The type of the object to be created.
+##	</summary>
+## </param>
+## <param name="object">
+##	<summary>
+##	The object class of the object being created.
+##	</summary>
+## </param>
+## <param name="name" optional="true">
+##	<summary>
+##	The name of the object being created.
+##	</summary>
+## </param>
+#
+interface(`init_filetrans_unit',`
+	gen_require(`
+		type systemd_unit_t;
+	')
+	allow $1 systemd_unit_t:dir search_dir_perms;
+	allow $1 systemd_unit_t:lnk_file read_lnk_file_perms;
+	filetrans_pattern($1, systemd_unit_t, $2, $3, $4)
+')
```

Comment:
I don't think I can go with this since it technically affects any of the system unit dir locations, not just the runtime unit dir, as you mention in the summary. Perhaps we should create a new runtime unit type.

What do you think @doverride

Reply:
Yes, I think it does make sense to setup a 'systemd_runtime_unit_t' or similar type for these files.

Reply:
This has been removed.  I realized there was an entry in the file_contexts.subs_dist making /run/systemd/system labeled list /lib/systemd/system.  Once I removed that I was able to use existing interface init_runtime_filetrans, hopefully this is more reasonable now.

Reply:
Sorry for late reply, yes probably best to create a special type for runtime unit files.